de_web_plugin.cpp

```diff
@@ -4535,10 +4537,24 @@ void DeRestPluginPrivate::checkSensorButtonEvent(Sensor *sensor, const deCONZ::A
                 }
                 else if (ind.clusterId() == SCENE_CLUSTER_ID && zclFrame.commandId() == 0x05) // recall scene
                 {
-                    ok = false; // payload: groupId, sceneId
-                    if (zclFrame.payload().size() >= 3 && buttonMap.zclParam0 == zclFrame.payload().at(2))
+                    ok = false; // payload: groupId (uint16) , sceneId (uint8)
+                    if (zclFrame.payload().size() >= 3)
                     {
-                        ok = true;
+                        // This device use same sceneID but different groupID : EDFF010000 ECFF010000 EBFF010000 EAFF010000
```

Comment:
Perhaps they got the order wrong as *[ sceneId, groupId ]*.
Do we have logs showing the destination group id in the APS indication?

Reply:
Will check, but it s same for the home/away switch (not included in deconz, too dangerous)

>     From Master remote SW Home / Away
>     Away : GroupId 0xfff6, SceneId 0x01
>     Home : GroupId 0xfff7, SceneId 0x01
>     Wakeup: GroupdId: 0xfff5, SceneId 0x01
>     Going to Sleep: GroupId: 0xfff4, SceneId 0x01

And no, I don't found logs, have just the payload, but have interesting message about group

> 10:35:00:164 [INFO] - Button 4002 - Pocket remote, broadcast to: 0xFFED, endpoint: 0x01, cluster: SCENES (0x0005), action: Scene 4, payload: EDFF010000, zclSeq: 3
> 10:35:00:170 delete old group 6 of sensor Pocket remote 2
> 10:35:00:171 attach group 65517 to sensor 2
> 10:35:00:353 Device TTL 3109 s flags: 0x7
> 10:35:01:468 [INFO] - Button 3002 - Pocket remote, broadcast to: 0xFFEC, endpoint: 0x01, cluster: SCENES (0x0005), action: Scene 3, payload: ECFF010000, zclSeq: 4
> 10:35:01:474 delete old group 65517 of sensor Pocket remote 2
> 10:35:01:475 create group 65516 for sensor 2
> 10:35:03:028 [INFO] - Button 2002 - Pocket remote, broadcast to: 0xFFEB, endpoint: 0x01, cluster: SCENES (0x0005), action: Scene 2, payload: EBFF010000, zclSeq: 5
> 10:35:03:038 delete old group 65516 of sensor Pocket remote 2
> 10:35:03:039 create group 65515 for sensor 2

Reply:
Currently the code doesn't expect a sensor to manage multiple groups that's why the logs show the "delete old group" messages.

Not sure if I fully understand the approach but I'd expect that the device inits different groups on when pairing. Having the group id values fixed in `button_maps.json` shouldn't work here? If the groups are always the same it would be strange as well, or does the plugin setup these via binding?

Reply:
Groups are always same. For all users, and fixed by devices, for exemple too (from pipiche github https://github.com/pipiche38/Domoticz-Zigate-Wiki/blob/master/en-eng/Corner_Legrand-Netatmo.md)

This is what happen when including a switch

>     Add Group 0xfff6
>         Add Scene 0x01 / Cluster 0x0006
>         Add Scene 0x01 / Cluster 0x0008
>     Add Group 0xfff4
>         Add Scene 0x01 / Cluster 0x0006
>         Add Scene 0x01 / Cluster 0x0008
>     Bind 0x0006
>     Configure Report 0x0006 / 0x0001 (Bind and Configure Reporting on 0x0008, will be done when enabling dimmer)
>     Bind 0x000f
>     Bind 0x0003

For information Legrand have too manufacture command using the Legrand cluster to manage group, but I don't use them for the moment.

Reply:
Have checked z2m, and they are doing same (for the home/away)

```
    legrand_scenes: {
        cluster: 'genScenes',
        type: 'commandRecall',
        convert: (model, msg, publish, options, meta) => {
            const lookup = {0xfff7: 'enter', 0xfff6: 'leave', 0xfff4: 'sleep', 0xfff5: 'wakeup'};
            return {action: lookup[msg.data.groupid] ? lookup[msg.data.groupid] : 'default'};
        },
    },
```

They use groups values

> From Master remote SW Home / Away
> Away : GroupId 0xfff6, SceneId 0x01
> Home : GroupId 0xfff7, SceneId 0x01
> Wakeup: GroupdId: 0xfff5, SceneId 0x01
> Going to Sleep: GroupId: 0xfff4, SceneId 0x01
>

Reply:
Interesting, yeah I image having multiple of these in the same network might be challenging or even clashes with other groups which by happen to use the same group id :/

Would be good to figure out how to query/set the used groups in future to have better control.

>It can be a problem ? I can block them in the code, but not sure it will be usefull ? For me it just provoke useless database IO, but this device is not used 10 time by day ....

Yes I think binding and the above group change checks would need to be disabled for this device.
It would be so much easier if they just use proper scene id's or even different endpoints :) (maybe scene ID can be configured?)

Reply:
>Would be good to figure out how to query/set the used groups in future to have better control.

This device is new, so can't be sure, but if it work like the home/away, groups are fixed by hardware, they are the same used on all zigbee project.

>maybe scene ID can be configured?

I don't think, I m checking my doc, and have only request to change groupe ID

>Yes I think binding and the above group change checks would need to be disabled for this device.

For group change, can add a bypass here
```
    else if (e.what() == REventValidGroup)
    {
```

But if I totaly disable binding, for exemple with blocking the "RConfigGroup" creation , the device can work ?

Reply:
Hmm not sure, at least we shouldn't try create own groups when the switch only uses the fixed groups anyway :)

Reply:
Yep, we will make some tries before.